actors/builtin/miner/miner_actor.go

```diff
@@ -902,85 +1025,75 @@ func (a Actor) ConfirmSectorProofsValid(rt Runtime, params *builtin.ConfirmSecto
 	// Ideally, we'd combine some of these operations, but at least we have
 	// a constant number of them.
 
-	var st State
-	rt.StateReadonly(&st)
-	store := adt.AsStore(rt)
-	info := getMinerInfo(rt, &st)
-
-	//
-	// Activate storage deals.
-	//
-
-	// This skips missing pre-commits.
-	precommittedSectors, err := st.FindPrecommittedSectors(store, params.Sectors...)
-	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pre-committed sectors")
-
 	// Committed-capacity sectors licensed for early removal by new sectors being proven.
 	replaceSectors := make(DeadlineSectorMap)
+	activation := rt.CurrEpoch()
 	// Pre-commits for new sectors.
-	var preCommits []*SectorPreCommitOnChainInfo
-	for _, precommit := range precommittedSectors {
-		if len(precommit.Info.DealIDs) > 0 {
+	var validPreCommits []*SectorPreCommitOnChainInfo
+	for _, preCommit := range preCommits {
+		duration := preCommit.Info.Expiration - activation
+		// This should have been caught in precommit, but don't let other sectors fail because of it.
+		if duration < MinSectorExpiration {
+			rt.Log(rtt.WARN, "precommit %d has lifetime %d less than minimum. ignoring", preCommit.Info.SectorNumber, duration, MinSectorExpiration)
+			continue
+		}
+
+		if len(preCommit.Info.DealIDs) > 0 {
 			// Check (and activate) storage deals associated to sector. Abort if checks failed.
 			// TODO: we should batch these calls...
 			// https://github.com/filecoin-project/specs-actors/issues/474
 			code := rt.Send(
 				builtin.StorageMarketActorAddr,
 				builtin.MethodsMarket.ActivateDeals,
 				&market.ActivateDealsParams{
-					DealIDs:      precommit.Info.DealIDs,
-					SectorExpiry: precommit.Info.Expiration,
+					DealIDs:      preCommit.Info.DealIDs,
```

Comment:
I suggest that changing capitalization style here isn't worthwhile, unless you're going to do so throughout the whole file (in another PR). We could debate the capitalisation but I'd end up agreeing with you so long as it was consistent everywhere.